src/engine/SCons/Environment.py

```diff
@@ -2305,7 +2305,19 @@ def __init__(self, subject, overrides={}):
 
     # Methods that make this class act like a proxy.
     def __getattr__(self, name):
-        return getattr(self.__dict__['__subject'], name)
+        attr = getattr(self.__dict__['__subject'], name)
+        if isinstance(attr, MethodWrapper):
```

Comment:
Stick some comments in hear to explain the why?

Reply:
Done.